pkg/agent/cniserver/interface_configuration_linux_test.go

```diff
@@ -0,0 +1,857 @@
+					fakeSriovNet.EXPECT().GetUplinkRepresentor(tc.sriovVFDeviceID).Return(sriovUplinkName, nil).Times(1)
+					fakeSriovNet.EXPECT().GetVfIndexByPciAddress(tc.sriovVFDeviceID).Return(sriovVfIndex, nil).Times(1)
+					fakeSriovNet.EXPECT().GetVfRepresentor(sriovUplinkName, sriovVfIndex).Return(sriovVfRepresentor, nil).Times(1)
+					if tc.renameIntefaceErr == nil {
+						hostInterfaceLink := &netlink.Dummy{
+							LinkAttrs: netlink.LinkAttrs{Index: 2, MTU: mtu, HardwareAddr: hostVethMac, Name: hostIfaceName, Flags: net.FlagUp},
+						}
+						fakeNetlink.EXPECT().LinkByName(hostIfaceName).Return(hostInterfaceLink, nil).Times(1)
+						moveVFtoNS = true
+					}
+				}
+			}
+			if tc.podSriovVFDeviceID != "" {
+				fakeSriovNet.EXPECT().GetPfName(tc.podSriovVFDeviceID).Return(sriovPfName, nil).Times(1)
+				fakeSriovNet.EXPECT().GetVfid(tc.podSriovVFDeviceID, sriovPfName).Return(sriovVfIndex, nil).Times(1)
+				fakeSriovNet.EXPECT().GetVFLinkNames(tc.podSriovVFDeviceID).Return(sriovVfNetdeviceName, nil).Times(1)
+				fakeNetlink.EXPECT().LinkByName(sriovVfNetdeviceName).Return(vfDeviceLink, nil).Times(1)
+				moveVFtoNS = true
+			}
+			if moveVFtoNS {
+				fakeNetlink.EXPECT().LinkByName(sriovVfNetdeviceName).Return(vfDeviceLink, nil).Times(1)
+				fakeNetlink.EXPECT().LinkSetNsFd(vfDeviceLink, gomock.Any()).Return(nil).Times(1)
+				containerInterfaceLink := &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Index: 2, MTU: mtu, HardwareAddr: containerVethMac, Name: containerIfaceName, Flags: net.FlagUp}}
+				fakeNetlink.EXPECT().LinkByName(containerIfaceName).Return(containerInterfaceLink, nil).Times(1)
+				fakeNetlink.EXPECT().LinkSetMTU(containerInterfaceLink, gomock.Any()).Return(nil).Times(1)
+				fakeNetlink.EXPECT().LinkSetUp(containerInterfaceLink).Return(nil).Times(1)
+			}
+			err := testIfConfigurator.configureContainerLink(podName, testPodNamespace, podContainerID, containerNS.Path(), containerIfaceName, mtu, tc.sriovVFDeviceID, tc.podSriovVFDeviceID, ipamResult, nil)
+			if tc.expectErr != nil {
+				assert.Error(t, err)
+				assert.Equal(t, tc.expectErr, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			err = containerNS.(*fakeNS).clear()
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestAdvertiseContainerAddr(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+
+	interfaceID := 1
+	ipv4CIDR := net.IPNet{
+		IP:   net.ParseIP("192.168.100.100"),
+		Mask: net.IPv4Mask(255, 255, 255, 0),
+	}
+	ipv4Gateway := net.ParseIP("192.168.100.1")
+	ipv6CIDR := net.IPNet{
+		IP:   net.ParseIP("fe12:ab::64:64"),
+		Mask: net.IPMask([]byte{255, 255, 255, 255, 255, 255, 255, 255, 0, 0, 0, 0, 0, 0, 0, 0}),
+	}
+	ipv6Gateway := net.ParseIP("fe12:ab::64:1")
+	defer mockIsNSorErr()()
+	defer mockWithNetNSPath()()
+	testIfConfigurator := newTestIfConfigurator(false, nil, nil)
+
+	for _, tc := range []struct {
+		name              string
+		result            *current.Result
+		runInNS           bool
```

Comment:
why pass it if it's always true?

Reply:
No, in case "result-no-ips", this is not required.